### PR TITLE
Adding distributions over eta and phi

### PR DESCRIPTION
**Pull request by natalia-korneeva**
Following last discussions with btagging@HLT conveners the eta/phi dimension's are to be monitored for future campaigns.

### CODE REVIEW

**Comment by natalia-korneeva** [comment]
New distributions for the 9_0_0_2016 release are available here 
https://twiki.cern.ch/twiki/bin/view/Sandbox/NataliaTsirovaSandbox#New_distributions_over_eta_and_p
They are Efficiency vs eta and Efficiency vs phi for b, c, light quarks.

**Comment by dmitrijus** [comment]
+1

File: HLTriggerOffline/Btag/src/HLTBTagHarvestingAnalyzer.cc [hunk 1]
```diff
@@ -46,13 +46,29 @@ HLTBTagHarvestingAnalyzer::dqmEndJob(DQMStore::IBooker & ibooker, DQMStore::IGet
 				efficsOK[flavour]=isOK;
 			}
 			label= m_histoName.at(ind)+std::string("___");
+			TString labelEta = label;
```
Comment (davidlange6) [inline HLTriggerOffline/Btag/src/HLTBTagHarvestingAnalyzer.cc]: Hi @natalia-korneeva - as not to make the existing issues with string conversions in this class worse, please change the types of labelEta and labelPhi to be std::string. That will avoid several conversions between string<->Tstring in the code you've added below.

File: HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc [hunk 2]
```diff
@@ -189,11 +197,19 @@ void HLTBTagPerformanceAnalyzer::bookHistograms(DQMStore::IBooker & ibooker, edm
 		int nBinsPt=60;
 		double pTmin=30;
 		double pTMax=330;
+		int nBinsPhi=54;
+		double phimin=-3.14;
```
Comment (davidlange6) [inline HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc]: could you use M_PI here?

**Comment by davidlange6** [comment]
-1
pending answers to small comments

**Comment by natalia-korneeva** [comment]
PR updated according to requests.

**Comment by natalia-korneeva** [comment]
The original PR to 900 is closed (https://github.com/cms-sw/cmssw/pull/18393).

**Comment by heppye** [comment]
green light from BTV@HLT. @dmitrijus @davidlange6, could u please take a look and sign again?

**Comment by dmitrijus** [comment]
+1

File: HLTriggerOffline/Btag/src/HLTBTagHarvestingAnalyzer.cc [hunk 3]
```diff
@@ -46,13 +46,29 @@ HLTBTagHarvestingAnalyzer::dqmEndJob(DQMStore::IBooker & ibooker, DQMStore::IGet
...
 			label+=flavour+TString("_disc_pT");
+			labelEta+=flavour+TString("_disc_eta");
+			labelPhi+=flavour+TString("_disc_phi");
 			isOK=GetNumDenumerators (ibooker,igetter,(TString(dqmFolder_hist)+"/"+label).Data(),(TString(dqmFolder_hist)+"/"+label).Data(),num,den,1);
 			if (isOK) {
 			
 				//do the 'b-tag efficiency vs pT' plot
 				TH1F eff=calculateEfficiency1D(ibooker,igetter,*num,*den,(label+"_efficiency_vs_pT").Data());
 			}
+			isOK=GetNumDenumerators (ibooker,igetter,(TString(dqmFolder_hist)+"/"+labelEta).Data(),(TString(dqmFolder_hist)+"/"+labelEta).Data(),num,den,2);
```
Comment (davidlange6) [inline HLTriggerOffline/Btag/src/HLTBTagHarvestingAnalyzer.cc]: no need for this TString or .Data() - just

dqmFolder_hist+"/"+labelEta,

File: HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc [hunk 4]
```diff
@@ -154,8 +154,14 @@ void HLTBTagPerformanceAnalyzer::analyze(const edm::Event& iEvent, const edm::Ev
 					H1_.at(ind)[label.Data()]->Fill(std::fmax(0.0,BtagJT.second));	//fill 1D btag plot for 'b,c,uds'
 					label=JetTagCollection_Label[ind] + "___";
 					label+=flavour_str;
+					TString labelEta = label;
```
Comment (davidlange6) [inline HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc]: make these std::string type

File: HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc [hunk 5]
```diff
@@ -154,8 +154,14 @@ void HLTBTagPerformanceAnalyzer::analyze(const edm::Event& iEvent, const edm::Ev
...
+					TString labelPhi = label;
 					label+=TString("_disc_pT");
 					H2_.at(ind)[label.Data()]->Fill(std::fmax(0.0,BtagJT.second),BtagJT.first->pt());	//fill 2D btag, jetPt plot for 'b,c,uds'
+					labelEta+=TString("_disc_eta");
+					H2Eta_.at(ind)[labelEta.Data()]->Fill(std::fmax(0.0,BtagJT.second),BtagJT.first->eta());	//fill 2D btag, jetEta plot for 'b,c,uds'
```
Comment (davidlange6) [inline HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc]: then no need for .Data()

File: HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc [hunk 6]
```diff
@@ -202,12 +218,22 @@ void HLTBTagPerformanceAnalyzer::bookHistograms(DQMStore::IBooker & ibooker, edm
 				H1_.back()[label.Data()] = 		 ibooker.book1D(label.Data(),   Form("%s %s",JetTagCollection_Label[ind].c_str(),flavour.Data()), btagBins, btagL, btagU );
 				H1_.back()[label.Data()]->setAxisTitle("disc",1);
 				label=JetTagCollection_Label[ind]+"___";
+				labelEta=label;
+				labelPhi=label;
 				label+=flavour+TString("_disc_pT");
+				labelEta+=flavour+TString("_disc_eta");
```
Comment (davidlange6) [inline HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc]: similarly here. - just +"_disc_eta"

**Comment by natalia-korneeva** [comment]
PR updated according to requests.

**Comment by mtosi** [comment]
as said, and agreed at the TSG meeting today
please,
1. add such feature in the code used by the DQM step
2. add plots for the HEP17, HEM17 and HEP18 slices
3. add ratio plots between HEP17, HEM17 and HEP18 (probably by using the GenericDQMClient)

**Comment by natalia-korneeva** [comment]
@mtosi, as discussed today at BTV@HLT meeting we propose to leave this PR as it is to be able to monitor the full eta and phi distributions. 
The HEP17 etc. plots are to be added with the separate PR as a separate task, this part is in preparation now.

**Comment by mtosi** [comment]
thanks for the clarification, hoping it will come soon ;)

**Comment by heppye** [comment]
@mtosi , thanks for the quick comment and hope this separate PRs plan is fine with release managements. @dmitrijus @davidlange6 , could u please have a look and sign off again? Thanks.

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**